### PR TITLE
test(instr-express): add test for ESM

### PR DESCRIPTION
**Pull request by JamieDanielson**
## Which problem is this PR solving?

- updates #1942 

## Short description of the changes

- add test for ESM support in express instrumentation

### CODE REVIEW

**Review by pichlermarc — APPROVED** [review]
Looks good, thank you for working on this. :slightly_smiling_face:

File: plugins/node/opentelemetry-instrumentation-express/test/express.test.ts [hunk 1]
```diff
@@ -493,4 +494,44 @@ describe('ExpressInstrumentation', () => {
...
+        assert.strictEqual(spans[1].name, 'middleware - query');
+        assert.strictEqual(spans[1].kind, SpanKind.SERVER);
+        assert.strictEqual(spans[1].parentSpanId, spans[0].spanId);
+        assert.strictEqual(spans[2].name, 'middleware - expressInit');
+        assert.strictEqual(spans[2].kind, SpanKind.SERVER);
+        assert.strictEqual(spans[2].parentSpanId, spans[0].spanId);
+        assert.strictEqual(spans[3].name, 'middleware - simpleMiddleware');
+        assert.strictEqual(spans[3].kind, SpanKind.SERVER);
+        assert.strictEqual(spans[3].parentSpanId, spans[0].spanId);
+        // assert.strictEqual(spans[4].name, 'router - /post/:id'); // TODO bug?
```
Comment (trentm) [inline plugins/node/opentelemetry-instrumentation-express/test/express.test.ts:530]: Yah, I think this is a bug, or limitation. I don't think the current express instrumentation handles usage of a `express.Router()`.   Perhaps change `use-express.mjs` for now to not use a Router and we add a separate issue about handling `Router` in instr-express?

File: plugins/node/opentelemetry-instrumentation-express/test/fixtures/use-express.mjs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  // can be flaky.
+  await promisify(setTimeout)(10);
+  next();
+});
+
+const router = express.Router();
+router.get('/post/:id', (req, res) => {
+  res.send(`Post id: ${req.params.id}`);
+});
+app.use(router);
```
Comment (trentm) [inline plugins/node/opentelemetry-instrumentation-express/test/fixtures/use-express.mjs:53]: As mentioned above, perhaps change this to:

```suggestion
app.get('/post/:id', (req, res) => {
  res.send(`Post id: ${req.params.id}`);
});
```

**Review by trentm — APPROVED** [review]
Thanks! LGTM, with a possible suggestion to avoid the current Router issue/limitation.

File: plugins/node/opentelemetry-instrumentation-express/test/express.test.ts (hunk 1 above)
Comment (JamieDanielson) [reply plugins/node/opentelemetry-instrumentation-express/test/express.test.ts:530]: I agree, will update

Comment (JamieDanielson) [reply plugins/node/opentelemetry-instrumentation-express/test/express.test.ts]: Updated this test and created new issue for tracking #1993

**Pull request merged**